### PR TITLE
added the sign up feature

### PR DESCRIPTION
**Pull request by tobemari**

### CODE REVIEW

File: src/Register.jsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    const [errMsg, setErrMsg] = useState('');
+    const [success, setSuccess] = useState(false);
+
+    useEffect(() => {
+        userRef.current.focus();
+    }, [])
+
+    useEffect(() => {
+        const result = USER_REGEX.test(user);
+        console.log(result);
```
Comment (akhilwaghmare) [inline src/Register.jsx]: Cleanup/remove console logs

File: src/Register.jsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        console.log(user);
+        setValidName(result);
+    }, [user])
+
+    useEffect(() => {
+        const result = PWD_REGEX.test(pwd);
+        console.log(result);
```
Comment (akhilwaghmare) [inline src/Register.jsx]: Cleanup/remove console logs

File: src/Register.jsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        console.log(pwd);
+        setValidPwd(result);
+        const match = pwd === matchPwd;
```
Comment (akhilwaghmare) [inline src/Register.jsx]: Bug: green checkmark shows next to confirm password even if original password is not valid.

File: src/index.css [hunk 4]
```diff
@@ -0,0 +1,29 @@
+
+.offscreen {
```
Comment (akhilwaghmare) [inline src/index.css]: Instead of this, can you just set `display: none` ?

File: src/Register.jsx (hunk 1 above)
Comment (tobemari) [reply src/Register.jsx]: Done! Thank you

File: src/Register.jsx (hunk 2 above)
Comment (tobemari) [reply src/Register.jsx]: Done! Thank you

File: src/Register.jsx (hunk 3 above)
Comment (tobemari) [reply src/Register.jsx]: Fixed

File: src/index.css (hunk 4 above)
Comment (tobemari) [reply src/index.css]: I will use screen reader for accessibility later so maybe it's better to use .offscreen?

File: src/pages/Register.jsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
+import { useRef, useState, useEffect, useReducer } from 'react';
```
Comment (akhilwaghmare) [inline src/pages/Register.jsx]: Nit: Remove `useReducer` if not used

File: src/pages/Register.jsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        {success ? (
+                <section>
+                    <h1>Success!</h1>
+                    <p>
+                        <a href="#">Sign In</a>
+                    </p>
+                </section>
+            ) : (
+                <section>
+                    <p ref={errRef} className={errMsg ? "errmsg" : "offscreen"} aria-live="assertive">{errMsg}</p>
```
Comment (akhilwaghmare) [inline src/pages/Register.jsx]: Instead of using the `offscreen` class, you can simplify by having the `<p>` tag conditionally render based on the value of `errMsg`

File: src/pages/Register.jsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+                            ref={userRef}
+                            autoComplete="off"
+                            onChange={(e) => setUser(e.target.value)}
+                            required
+                            aria-invalid={validName ? "false" : "true"}
+                            aria-describedby="uidnote"
+                            onFocus={() => setUserFocus(true)}
+                            onBlur={() => setUserFocus(false)}
+                        />
+                        <p id="uidnote" className={userFocus && user && !validName ? "instructions" : "offscreen"}>
```
Comment (akhilwaghmare) [inline src/pages/Register.jsx]: Same as above - can make conditional rendering instead of different classes

File: src/pages/Register.jsx [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+                            type="password"
+                            id="password"
+                            onChange={(e) => setPwd(e.target.value)}
+                            required
+                            aria-invalid={validPwd ? "false" : "true"}
+                            aria-describedby="pwdnote"
+                            onFocus={() => setPwdFocus(true)}
+                            onBlur={() => setPwdFocus(false)}
+                        />
+                        <p id="pwdnote" className={pwdFocus && !validPwd ? "instructions" : "offscreen"}>
```
Comment (akhilwaghmare) [inline src/pages/Register.jsx]: Same as above - can make conditional rendering instead of different classes

File: src/pages/Register.jsx [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+                            type="password"
+                            id="confirm_pwd"
+                            onChange={(e) => setMatchPwd(e.target.value)}
+                            required
+                            aria-invalid={validMatch ? "false" : "true"}
+                            aria-describedby="confirmnote"
+                            onFocus={() => setMatchFocus(true)}
+                            onBlur={() => setMatchFocus(false)}
+                        />
+                        <p id="confirmnote" className={matchFocus && !validMatch ? "instructions" : "offscreen"}>
```
Comment (akhilwaghmare) [inline src/pages/Register.jsx]: Same as above - can make conditional rendering instead of different classes

File: src/pages/Register.jsx (hunk 7 above)
Comment (akhilwaghmare) [inline src/pages/Register.jsx]: Think about if there's a way to consolidate all the "instructions" content into a single `<p>` tag with dynamic content based on validation and which input is in focus

File: src/pages/Register.jsx [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+                    <h1>Success!</h1>
+                    <p>
+                        <a href="#">Sign In</a>
+                    </p>
+                </section>
+            ) : (
+                <section>
+                    <p ref={errRef} className={errMsg ? "errmsg" : "offscreen"} aria-live="assertive">{errMsg}</p>    
+                    
+                    <form onSubmit={handleSubmit}>
```
Comment (akhilwaghmare) [inline src/pages/Register.jsx:134]: Simple style improvement: stack the inputs vertically. Can group each label and input together, then use flexbox with direction of column

**Review by akhilwaghmare — CHANGES_REQUESTED** [review]
Left some requested changes and comments to improve code quality.

File: src/pages/Register.jsx (hunk 5 above)
Comment (tobemari) [reply src/pages/Register.jsx]: Done

File: src/pages/Register.jsx (hunk 6 above)
Comment (tobemari) [reply src/pages/Register.jsx]: Done

File: src/pages/Register.jsx (hunk 7 above)
Comment (tobemari) [reply src/pages/Register.jsx]: Done

File: src/pages/Register.jsx (hunk 8 above)
Comment (tobemari) [reply src/pages/Register.jsx]: Done

File: src/pages/Register.jsx (hunk 10 above)
Comment (tobemari) [reply src/pages/Register.jsx:134]: Done

File: src/pages/Register.jsx (hunk 7 above)
Comment (tobemari) [reply src/pages/Register.jsx]: done

File: src/pages/Register.jsx (hunk 10 above)
Comment (tobemari) [reply src/pages/Register.jsx:68]: Done

File: src/pages/LogIn.jsx [hunk 11]
```diff
@@ -1,14 +1,76 @@
...
+const handleSubmit = async (e) => {
+  e.preventDefault();
+  console.log(user, pwd);
+  setUser('');
+  setPwd('');
+  setSuccess(true);
+}
   return (
+  <>
+    {success ? (
```
Comment (akhilwaghmare) [inline src/pages/LogIn.jsx]: What is this `success` variable? It isn't defined anywhere in this file

File: src/Users.jsx
Comment (akhilwaghmare) [inline src/Users.jsx:1]: Can you help me understand what this page is meant to do? Why are we showing the list of all users?

File: package.json [hunk 12]
```diff
@@ -4,17 +4,22 @@
   "version": "0.0.0",
   "type": "module",
   "scripts": {
+    "start": "node app.js",
```
Comment (akhilwaghmare) [inline package.json]: Why was this added? There is no `app.js` file

File: src/api/axios.js [hunk 13]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
+
+export default axios.create({
+    baseURL: 'http://localhost:8000/api/v1'
```
Comment (akhilwaghmare) [inline src/api/axios.js]: Can we make the base part of this (`http://localhost:8000`) an environment variable? That way we can switch between the local version of the backend and the deployed version easily without adjusting the code

File: src/pages/Register.jsx [hunk 14]
```diff
@@ -0,0 +1,281 @@
...
+                            <a href="/login">Sign In</a>
+                        </span>
+                    </p>
+                </section>
+            )}
+        </div>
+    )
+}
+
+await axios.post(REGISTER_URL, {
```
Comment (akhilwaghmare) [inline src/pages/Register.jsx]: Why are we passing hardcoded values to the API, and not the values from the input fields?

**Review by akhilwaghmare — CHANGES_REQUESTED** [review]
Left a few comments throughout. Also, noticed there are some merge conflicts with the `main` branch, let me know if you need help sorting through those.

File: src/Users.jsx
Comment (tobemari) [reply src/Users.jsx:1]: I created it before our Monday meeting, it supposed to be the component for the admin page. On Monday we decided that we don't need admin page so I should remove it.

File: src/pages/LogIn.jsx (hunk 11 above)
Comment (tobemari) [reply src/pages/LogIn.jsx]: My bad, I wanted to push just Register.jsx, not all changes. I am still working on LogIn.jsx, so it's not ready to rewiew

File: package.json (hunk 12 above)
Comment (tobemari) [reply package.json]: Deleted

File: src/pages/Register.jsx (hunk 14 above)
Comment (tobemari) [reply src/pages/Register.jsx]: It was test data, now removed. I am working on it for the next commit

File: src/api/axios.js (hunk 13 above)
Comment (tobemari) [reply src/api/axios.js]: done

**Review by akhilwaghmare — APPROVED** [review]
